File: BE/src/main/java/com/haivn/common_api/KeHoachThucTap.java
package com.haivn.common_api;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.DynamicUpdate;
import org.hibernate.annotations.Where;

import javax.persistence.*;
import java.sql.Timestamp;

@Entity
@Table(name = "ke_hoach_thuc_tap")
@Getter
@Setter
@DynamicUpdate
@Where(clause = "deleted=false")
public class KeHoachThucTap extends BaseEntity{
    @Column(name = "tilte")
    private String tilte;
    @Column(name = "content")
    private String content;
    @Column(name = "statr_date")
    private Timestamp startDate;
    @Column(name = "han_dang_ky")
    private Timestamp hanDangKy;
    @Column(name = "ngay_di_tt")
    private Timestamp ngayDiTt;
    @Column(name = "ngay_ve_tt")
    private Timestamp ngayVeTt;
    @Column(name = "han_nop_bao_cao")
    private Timestamp hanNopBaoCao;
    @Column(name = "end_date")
    private Timestamp endDate;
    @Column(name = "file_danh_sach")
    private String fileDanhSach;
    @Column(name = "status")
    private Short status;
}

File: fe_web/lib/models/ke-hoach-thuc-tap.dart
// ignore_for_file: file_names

class KeHoachThucTap {
  int? id;
  String? tilte;
  String? content;
  String? startDate;
  String? endDate;
  String? hanDangKy;
  String? hanNopBaoCao;
  String? ngayDiThuctap;
  String? ngayHetThuctap;
  String? fileDanhSach;
  int? status;
  KeHoachThucTap({
    this.id,
    this.tilte,
    this.content,
    this.startDate,
    this.endDate,
    this.hanDangKy,
    this.hanNopBaoCao,
    this.ngayDiThuctap,
    this.ngayHetThuctap,
    this.fileDanhSach,
    this.status,
  });

  KeHoachThucTap.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        tilte = json['tilte'],
        content = json['content'],
        startDate = json['startDate'],
        endDate = json['endDate'],
        hanDangKy = json['hanDangKy'],
        ngayDiThuctap = json['ngayDiTt'],
        ngayHetThuctap = json['ngayVeTt'],
        hanNopBaoCao = json['hanNopBaoCao'],
        fileDanhSach = json['fileDanhSach'],
        status = json['status'];

  Map<String, dynamic> toJson() => {
        'tilte': tilte,
        'content': content,
        'startDate': startDate,
        'endDate': endDate,
        'hanDangKy': hanDangKy,
        'hanNopBaoCao': hanNopBaoCao,
        'ngayDiTt': ngayDiThuctap,
        'ngayVeTt': ngayHetThuctap,
        'fileDanhSach': fileDanhSach,
        'status': status
      };
}
